### PR TITLE
JsonResponse::__construct() gets charset as optional parameter

### PR DESCRIPTION
**Pull request by mkoubik**

### CODE REVIEW

**Comment by Majkl578** [comment]
:-1:, as per RFC 4627, JSON SHALL (must) be encoded in Unicode.

**Comment by mkoubik** [comment]
Cool, I didn't know.

**Comment by JanTvrdik** [comment]
@Majkl578 Unicode is not an encoding. JSON may be encoded in UTF-16.

So I'm :+1: on this RFC.

File: src/Application/Responses/JsonResponse.php [hunk 1]
```diff
@@ -26,18 +26,23 @@ class JsonResponse extends Nette\Object implements Nette\Application\IResponse
...
+
 
 	/**
 	 * @param  array|\stdClass  payload
-	 * @param  string    MIME content type
+	 * @param  string           MIME content type
+	 * @param  string           charset
 	 */
-	public function __construct($payload, $contentType = NULL)
+	public function __construct($payload, $contentType = NULL, $charset = NULL)
```
Comment (JanTvrdik) [inline src/Application/Responses/JsonResponse.php]: What about making UTF-8 a default `$charset` value?

**Comment by mkoubik** [comment]
@JanTvrdik:
```
Since the first two characters of a JSON text will always be ASCII
   characters [RFC0020], it is possible to determine whether an octet
   stream is UTF-8, UTF-16 (BE or LE), or UTF-32 (BE or LE) by looking
   at the pattern of nulls in the first four octets.
```

File: src/Application/Responses/JsonResponse.php (hunk 1 above)
Comment (JanTvrdik) [inline src/Application/Responses/JsonResponse.php]: Oh, I did not realize the due to using `json_encode` the encoding will always be UTF-8.

**Comment by Majkl578** [comment]
@JanTvrdik: I know, but I've never seen other than UTF-8 in real life so far, at least in PHP.

**Pull request closed without merging**